kikuchipy/signals/ebsd_master_pattern.py

```diff
@@ -289,3 +407,224 @@ class LazyEBSDMasterPattern(EBSDMasterPattern, LazySignal2D):
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+
+def _get_direction_cosines(detector: EBSDDetector) -> Vector3d:
+    """
+    Get the direction cosines between the detector and sample as done in EMsoft
+    and [Callahan2013]_.
+
+    Parameters
+    ----------
+    detector : EBSDDetector
+        EBSDDetector object with a certain detector geometry.
+
+    Returns
+    ----------
+    r_g.unit : Vector3d
+        Vector3d object containing the direction cosines for each detector
+        pixel.
+    """
+
+    pc = detector.pc_emsoft()
+    xpc = pc[..., 0]
+    ypc = pc[..., 1]
+    L = pc[..., 2]
+
+    # Detector coordinates in microns
+    det_x = (
+        -((-xpc - (1.0 - detector.ncols) * 0.5) - np.arange(0, detector.ncols))
+        * detector.px_size
+    )
+    det_y = (
+        (ypc - (1.0 - detector.nrows) * 0.5) - np.arange(0, detector.nrows)
+    ) * detector.px_size
+
+    # Auxilliary angle to rotate between reference frames
+    theta_c = np.radians(detector.tilt)
```

Comment:
Do you think it would be best to always use radians internally, and degrees only in `EBSDDetector.__repr__()`? I.e., so you don't have to convert to radians here?

Reply:
Internally, yes

Reply:
Okay, I think we'll keep this in mind for later.